src/rincewind_wizard.erl

```diff
@@ -0,0 +1,47 @@
+-module(rincewind_wizard).
+
+-behaviour(gen_server).
+
+-type name() :: atom().
+-type definition() :: #{name := name(), phases := [rincewind_phase:definition(), ...]}.
+
+-opaque t() ::
+    #{name := name(),
+      pid := pid(),
+      phases := [rincewind_phase:t(), ...]}.
```

Comment:
In `rince_wizard:definition()` the `phases` are `rincewind_phase:definition()`s whereas in the `rincewind_wizard:t()` type the phases are `rincewind_phase:t()`s. I suppose that this is considering that in the future the two types could be different (a phase definition might not exactly look like the internal phase type) but at the moment of writing they are equivalent (both are `#{name := name()}`).

Can we simplify this in favor of using one type until we need a second one? It would make things easier to follow, IMHO, and it could very well be the case that we never end up needing two different types.

Reply:
The moment when we need two types will be precisely my next PR ;)